Library/Services/Implementation/NfieldMediaFilesService.cs

```diff
@@ -110,6 +112,30 @@ public Task AddOrUpdateAsync(string surveyId, string fileName, byte[] content)
                       .FlattenExceptions();
         }
 
+        public Task UploadAndSaveAsync(string surveyId, string fileName, byte[] content)
+        {
+            if (string.IsNullOrEmpty(surveyId))
```

Comment:
Most services use a private Check method for checking the arguments. This would make the code a bit more readable.
See for instance NfieldSurveySampleDataService => CheckRequiredArgument method,
or in lots of services => CheckSurveyId method.